docs/book/book.typ

```diff
@@ -0,0 +1,178 @@
+
+- The ActorFramework, which provides concurrency and distribution primitives. Components like AddressableActor, EngineWorker, Exchange, and TypeFactory implement the actor model, enabling message-driven communication and fault-tolerant execution across nodes.
+
+- Finally, the Foundation layer provides essential runtime. The ZIO package offers structured concurrency, logging, metrics, and configuration management. The OTP package provides connectivity over erlang distribuition protocol. The OTP package uses `jInterface` package which provides low-level interoperability with Erlang nodes.
+
+Overall, this layered design separates concerns clearly: the Clouseau package focuses on search-specific logic, the ActorFramework provides concurrency and messaging, the Facade layer integrates these with the actor runtime, and the Foundation layer delivers robust primitives for process management and communication.
+
+#box(components-diagram)
+
+#pagebreak()
+
+= Inheritance diagram
+
+In Clouseau's architecture, all business logic classes share a common inheritance pattern that begins with the `Service` class and ultimately implements the `Actor` trait through an inheritance chain. At the foundation, the `ProcessLike` trait defines the essential behavior expected from any actor-like component. This trait is extended by the `Process` class, which introduces core attributes like runtime, name, self, and node. Building on `Process`, the `Service` class acts as the primary abstraction for all business logic components. It encapsulates common service-level responsibilities such as initialization, message handling, and graceful shutdown, and most importantly adaptiation between async and sync parts of the system. Every major component in Clouseau—such as ClouseauSupervisor, IndexManagerService, IndexService, AnalyzerService, InitService, and RexService—inherits from Service. Through this chain, each service class becomes an `AddressableActor` indirectly, because `Process` and `ProcessLike` integrate with the actor model provided by the underlying runtime. This is crucial for Clouseau's distributed nature: actors can be addressed, monitored, and linked across nodes. The inheritance hierarchy was chosen to refelct the same structure provided by Scalang (actor framework used by Clouseau 2x. By rooting all business logic in Service and connecting it to the actor system via Process and ProcessLike, Clouseau achieves a clean separation of concerns: the actor framework handles concurrency and distribution, while service classes implement domain-specific functionality. Keep in mind that as we rewrite services in async maner the inheritance hierarchy would change and eventually the `Service` would be removed, which would simplify the system a lot.
```

Comment:
At the mention of Scalang, the segment in parenthesis does not seem to have a closing bracket.